### PR TITLE
MINIFI-239: Add InvokeHTTP Processor and corresponding tests

### PR DESCRIPTION
**Pull request by phrocker**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by phrocker** [comment]
@apiri and @randerzander does not encompass all features for this ticket, but provides some basic functionality.

**Comment by apiri** [comment]
reviewing

File: libminifi/include/processors/InvokeHTTP.h [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+  /**
+   * Receive HTTP Response.
+   */
+  static size_t recieve_write(char * data, size_t size, size_t nmemb,
+                              void * p) {
+    return static_cast<HTTPRequestResponse*>(p)->write_content(data, size,
+                                                               nmemb);
+  }
+  /**
+     * Callback for post, put, and patch operations
```
Comment (apiri) [inline libminifi/include/processors/InvokeHTTP.h]: comments are not aligned with current impl

**Comment by apiri** [comment]
verifying updates

**Comment by apiri** [comment]
got caught up with some other things, but this looks good.  will get merged, thanks for the fixes

**Pull request closed without merging**